### PR TITLE
feat: Batch update points

### PR DESCRIPTION
**Pull request by Anush008**
This PR

- Adds support for [batch point update operation](https://qdrant.tech/documentation/concepts/points/#batch-update).
- Updates the `VectorsFactory.namedVector()` to accept `Map<String, Vector>` to support creating sparse vectors. Can't add an overload for generic types.

### CODE REVIEW

File: src/main/java/io/qdrant/client/VectorsFactory.java [hunk 1]
```diff
@@ -18,13 +19,13 @@ private VectorsFactory() {
...
-	 * @param values A map of vector names to values
+	 * @param values A map of vector names to {@link Vector}
 	 * @return a new instance of {@link Vectors}
 	 */
-	public static Vectors namedVectors(Map<String, List<Float>> values) {
+	public static Vectors namedVectors(Map<String, Vector> values) {
 		return Vectors.newBuilder()
 			.setVectors(NamedVectors.newBuilder()
-				.putAllVectors(Maps.transformValues(values, v -> vector(v)))
+				.putAllVectors(Maps.transformValues(values, v -> v))
```
Comment (russcam) [inline src/main/java/io/qdrant/client/VectorsFactory.java:28]:
```suggestion
				.putAllVectors(values)
```

File: src/main/java/io/qdrant/client/VectorsFactory.java [hunk 2]
```diff
@@ -18,13 +19,13 @@ private VectorsFactory() {
 
 	/**
 	 * Creates named vectors
-	 * @param values A map of vector names to values
+	 * @param values A map of vector names to {@link Vector}
 	 * @return a new instance of {@link Vectors}
 	 */
-	public static Vectors namedVectors(Map<String, List<Float>> values) {
+	public static Vectors namedVectors(Map<String, Vector> values) {
```
Comment (russcam) [inline src/main/java/io/qdrant/client/VectorsFactory.java:25]: What form of versioning is the team following for clients? This is an API breaking change in a non-prerelease versioned package, so should be handled according to the versioning scheme chosen.

At a minimum, I'd suggest calling this out as a breaking change in the next release notes.

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 3]
```diff
@@ -2204,6 +2207,65 @@ public ListenableFuture<List<BatchResult>> recommendBatchAsync(
...
+
+	/**
+	 * Performs a batch update of points.
+	 * 
+	 * @param request The update batch request.
+	 * @param timeout The timeout for the call.
+	 * 
+	 * @return a new instance of {@link ListenableFuture}
+	 */
+	public ListenableFuture<UpdateBatchResponse> batchUpdateAsync(UpdateBatchPoints request, @Nullable Duration timeout) {
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:2260]:
```suggestion
	public ListenableFuture<List<UpdateResult>> batchUpdateAsync(UpdateBatchPoints request, @Nullable Duration timeout) {
```

Would align with the convention of other batch APIs

File: src/main/java/io/qdrant/client/QdrantClient.java [hunk 4]
```diff
@@ -2204,6 +2207,65 @@ public ListenableFuture<List<BatchResult>> recommendBatchAsync(
...
+		String collectionName = request.getCollectionName();
+		Preconditions.checkArgument(!collectionName.isEmpty(), "Collection name must not be empty");
+		logger.debug("Batch update points on '{}'", collectionName);
+		ListenableFuture<UpdateBatchResponse> future = getPoints(timeout).updateBatch(request);
+		addLogFailureCallback(future, "Batch update points");
+		return future;
```
Comment (russcam) [inline src/main/java/io/qdrant/client/QdrantClient.java:2266]:
```suggestion
		return Futures.transform(
			future,
			UpdateBatchResponse::getResultList,
			MoreExecutors.directExecutor());
```

Would align with the convention of other batch APIs

File: src/main/java/io/qdrant/client/VectorsFactory.java (hunk 2 above)
Comment (Anush008) [reply src/main/java/io/qdrant/client/VectorsFactory.java:25]: The major and minor versions of clients are to be in parity with the Qdrant version, i.e. `1.7`. The patch versions can be bumped.

I will include the notice in the Release Notes.

**Comment by Anush008** [comment]
@russcam, can I get your thoughts on https://github.com/qdrant/java-client/pull/16/commits/2579be1af9a9c945dd928f5a8c32c842c9764761?

**Pull request merged**